TF refactor

katlapinka opened the pull request.

Commit:
Move size to types

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move Time to types

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move output to connection utils

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move asynchronous to connection utils

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move retry to connection utils

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move singleton to common utils

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move checksec to scripts

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move error device to storage devices as separate file

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move linux command and wait method to common tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move dmesg to tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Remove generator from test utils

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move fstab to tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move git to tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Move scsi_debug to tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Remove kedr

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Rename systemd to systemctl and move it to tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
OS tools refactor:
- make udev, runlevel, memory and wget separate files
- rename os_utils to os_tools and move it to test_tools
- remove ModuleRemoveMethod and set modprobe as default
- fix regex in is_kernel_module_loaded method
- remove get_sys_block_path
- move get_block_device_names method to disk tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Disk finder refactor

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Types fix - rename to type_def

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Iostat refactor

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Refactor disk tools and fs tools

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Commit:
Fix imports

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

Deixx commented on log/logger.py:
```diff
@@ -188,7 +188,7 @@ def step_info(self, step_name):
     def get_additional_logs(self):
         from core.test_run import TestRun
-        from test_tools.fs_utils import check_if_file_exists
+        from test_tools.fs_tools import check_if_file_exists
```
do we still need to import it here instead of at the top of the file?

katlapinka replied:
It needs to be imported here because fs_tools imports TestRun which imports logger and there we have circular dependencies. I think that core functionalities refactor to avoid circular dependencies will be done in next step in other PR

Deixx commented on storage_devices/device.py:
```diff
@@ -34,49 +36,49 @@ def is_mounted(self):
             TestRun.LOGGER.info("Device is not mounted.")
-        elif disk_utils.unmount(self):
+        elif unmount(self):
             self.mount_point = None
```
I'd prefer that methods don't look like they are calling themselves (I know they call imported methods, but at the first glance this can be confusing). For the sake of readability I'd use disk_tools.mount, disk_tools.unmount

Deixx commented on storage_devices/device.py:
```diff
@@ -34,49 +36,49 @@ def is_mounted(self):
-        from test_tools import fs_utils
-        output = fs_utils.ls(f"$(find -L {directory} -samefile {self.path})")
-        return fs_utils.parse_ls_output(output, self.path)
+        from test_tools import fs_tools
```
can it be moved to the top of the file?

Deixx commented on storage_devices/disk.py:
```diff
@@ -137,33 +140,33 @@ def resolve_type(cls, disk_path):
         part_number = int(part.path.split("part")[1])
-        disk_utils.remove_parition(self, part_number)
+        remove_parition(self, part_number)
         self.partitions.remove(part)
```
disk_tools.create_partitions, disk_tools.remove_partition?

Deixx commented on storage_devices/disk.py:
```diff
@@ -137,33 +140,33 @@ def resolve_type(cls, disk_path):
+            if is_mounted(part.path):
                 part.unmount()
-        if disk_utils.remove_partitions(self):
+        if remove_partitions(self):
```
disk_tools.remove_paritions?

Deixx commented on storage_devices/disk.py:
```diff
@@ -242,21 +279,21 @@ def unplug(self) -> Output:
 
     def get_lba_format_in_use(self):
-        return nvme_cli.get_lba_format_in_use(self)
+        return get_lba_format_in_use(self)
```
nvme_cli. ?

Deixx commented on storage_devices/disk.py:
```diff
@@ -311,8 +348,8 @@ def get_unplug_path(self, device_id) -> str:
         ls_command = f"$(find -H /sys/devices/ -name {device_id} -type d)"
-        output = fs_utils.ls_item(f"{ls_command}")
-        sysfs_addr = fs_utils.parse_ls_output(output)[0]
+        output = ls_item(f"{ls_command}")
```
ls_command is already a string

Deixx commented on storage_devices/disk.py:
```diff
@@ -378,8 +415,8 @@ def get_pci_address(cls, device_id) -> str:
         ls_command = f"$(find -H /sys/devices/ -name {device_id} -type d)"
-        output = fs_utils.ls_item(f"{ls_command}")
-        sysfs_addr = fs_utils.parse_ls_output(output)[0]
+        output = ls_item(f"{ls_command}")
```
same

Deixx commented on test_tools/dd.py:
```diff
@@ -3,8 +3,8 @@
 
-import test_utils.linux_command as linux_comm
-import test_utils.size as size
+import test_tools.common.linux_command as linux_comm
```
```suggestion
from test_tools.common.linux_command import LinuxCommand
```

Deixx commented on test_tools/ddrescue.py:
```diff
@@ -3,14 +3,14 @@
 
-import test_utils.linux_command as linux_comm
-import test_utils.size as size
+import type_def.size as size
```
```suggestion
from type_def.size import Size
```

Deixx commented on test_tools/fs_tools.py:
```diff
@@ -7,15 +7,23 @@
 from collections import namedtuple
 from datetime import datetime, timedelta
+from enum import Enum
+from aenum import IntFlag
```
can't we import from enum here?

katlapinka replied:
IntFlag from enum is not able to correctly parse string like "x|y|z"

Deixx commented on test_tools/peach_fuzzer/peach_fuzzer.py:
```diff
@@ -12,10 +12,10 @@
 import lxml.etree as etree
 from collections import namedtuple
 
+import test_tools.wget
```
```suggestion
from test_tools import wget
```

Deixx commented on test_utils/filesystem/directory.py:
```diff
@@ -13,14 +13,14 @@ def __init__(self, full_path):
-        fs_utils.create_directory(path, parents)
-        output = fs_utils.ls_item(path)
-        return fs_utils.parse_ls_output(output)[0]
+        create_directory(path, parents)
```
fs_tools.ls, fs_tools.create_directory ?

Deixx replied:
I assume self.full_path is already a string

Deixx commented on test_utils/filesystem/file.py:
```diff
@@ -3,72 +3,74 @@
 
     def crc32sum(self, timeout: timedelta = timedelta(minutes=30)):
-        return fs_utils.crc32sum(str(self), timeout)
+        return crc32sum(str(self), timeout)
```
and here?

Deixx commented on test_utils/filesystem/file.py:
```diff
@@ -3,72 +3,74 @@
-        fs_utils.create_file(path)
-        output = fs_utils.ls_item(path)
-        return fs_utils.parse_ls_output(output)[0]
+        create_file(path)
```
and here?

Deixx commented on test_utils/filesystem/file.py:
```diff
@@ -3,72 +3,74 @@
              timeout: timedelta = timedelta(minutes=30)):
-        fs_utils.copy(str(self), destination, force, recursive, dereference, timeout)
-        if fs_utils.check_if_directory_exists(destination):
+        copy(str(self), destination, force, recursive, dereference, timeout)
```
and here?

Deixx commented on test_utils/filesystem/fs_item.py:
```diff
@@ -38,40 +39,40 @@ def __str__(self):
-        target_dir_exists = fs_utils.check_if_directory_exists(destination)
-        fs_utils.move(str(self), destination, force)
+        target_dir_exists = check_if_directory_exists(destination)
+        move(str(self), destination, force)
```
and here?

Commit:
Add trim verification fio param

This param will be needed to replace some vdbench configs after removing
vdbench plugin

Signed-off-by: Katarzyna Treder <katarzyna.treder@h-partners.com>

pep8speaks commented:
Hello @katlapinka! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





























































































































There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2024-12-31 10:52:02 UTC
